asset/js/widget/BaseInput.js

```diff
@@ -666,77 +688,95 @@ define(["../notjQuery", "Completer"], function ($, Completer) {
-                            input = this.moveFocusForward(termIndex - 1);
-                            if (event.ctrlKey || event.metaKey) {
-                                this.clearPartialTerm(input);
-                            } else {
-                                this.writePartialTerm(input.value.slice(1), input);
+                        removedTerms = this.clearSelectedTerms();
+
+                        if (termIndex >= 0 && (! input.value || this.isTermDirectionVertical())) {
+                            let removedTerm = this.removeTerm(input.parentNode);
+                            if (removedTerm !== false) {
+                                input = this.moveFocusForward(termIndex - 1);
+                                if (event.ctrlKey || event.metaKey) {
+                                    this.clearPartialTerm(input);
+                                } else {
+                                    this.writePartialTerm(input.value.slice(1), input);
+                                }
+
+                                removedTerms[termIndex] = removedTerm;
+                                event.preventDefault();
                             }
-
-                            removedTerms[termIndex] = removedTerm;
-                            event.preventDefault();
                         }
-                    }
 
-                    this.togglePlaceholder();
-                    this.autoSubmit(input, 'remove', removedTerms);
+                        this.togglePlaceholder();
+                        this.autoSubmit(input, 'remove', removedTerms);
                     break;
                 case 'Enter':
                     if (termIndex >= 0) {
                         this.saveTerm(input, false);
                     }
                     break;
                 case 'ArrowLeft':
-                    if (input.selectionStart === 0 && this.hasTerms()) {
+                    if (! this.isTermDirectionVertical() && this.hasTerms()
+                        && (input.selectionStart === 0 || input.selectionStart === null && this.isReadOnlyMode())) {
                         event.preventDefault();
                         this.moveFocusBackward();
                     }
                     break;
                 case 'ArrowRight':
-                    if (input.selectionStart === input.value.length && this.hasTerms()) {
+                    if (! this.isTermDirectionVertical() && this.hasTerms()
+                        && (input.selectionStart === input.value.length || input.selectionStart === null && this.isReadOnlyMode())) {
+                        event.preventDefault();
+                        this.moveFocusForward();
+                    }
+                    break;
+                case 'ArrowUp':
+                    if (this.isTermDirectionVertical() && this.hasTerms()
+                        && (input.selectionStart === 0 || input.selectionStart === null && this.isReadOnlyMode())) {
```

Comment:
I see no reason for checking the cursor position here. It's checked for left and right because that's how the cursor can move inside the input. This is about up and down which are directions the cursor can't move inside a single line text input anyway.

But, you should check if the input is being completed or not. Currently this and the completer compete who focuses what if there are suggestions open. Add this condition: `this.completer === null || ! this.completer.isBeingCompleted(input)`

Reply:
Added `this.completer === null || ! this.completer.isBeingCompleted(input) && ! this.completer.hasSuggestions()`. 

! this.completer.hasSuggestions()` ensures that the suggestion list is not visible, otherwise pressing ArrowDown or ArrowUp to select the suggested options will change the focus of the terms. This will cause the selected suggestion to overwrite the existing term at position 0. 

Or if a term is being edited and something is selected from the suggestion list, a new duplicate term is added because the focus has moved again.